pkg/auth/cred.go

```diff
@@ -29,7 +30,13 @@ func NewCredential(cfg *Config) (azcore.TokenCredential, error) {
 		return nil, fmt.Errorf("failed to create credential, nil config provided")
 	}
 
+	if cfg.UseNewCredWorkflow {
```

Comment:
Can we name this something more fitting than NewCredWorkflow? Most auth code is ripped out by others and its probably the most likely candidate to get copy pasted from karpenter into another project, so naming moreso around workloadIdentity than UseNewCredWorkFlow makes sense to me

Reply:
Yea, this was just my general temp name, and happy to change, but didn't have a better one in mind.

I was thinking something like `UseWorkloadIdentity`. However, this path doesn't have to just be for workload identity is the issue, as the backing function I'm pretty sure could handle other cases too.

Reply:
UseWorkloadEntity makes a lot more sense to me, its not apparent how the backing function is being used for other cases.

Reply:
Since it's just calling the sdk default method, and AFAIK that default method is primarily based on checking the env for things such as:
```
	azureClientID                   = "AZURE_CLIENT_ID"
	azureClientSecret               = "AZURE_CLIENT_SECRET"
	azureFederatedTokenFile         = "AZURE_FEDERATED_TOKEN_FILE"
```

Maybe something around `UseCredentialFromEnvironment`? This path ought to work for more than just the workload identity flow, it works for SP too if you put the clientid/clientsecret into those env variables and pass them to this pod.

Reply:
Yea, I feel from env could be a good naming. Feels better than the others.

Reply:
Updated naming.